src/contracts/OriginARM.sol

```diff
@@ -0,0 +1,102 @@
+    constructor(
+        address _otoken,
+        address _liquidityAsset,
+        address _vault,
+        uint256 _claimDelay,
+        uint256 _minSharesToRedeem
+    ) AbstractARM(_liquidityAsset, _otoken, _liquidityAsset, _claimDelay, _minSharesToRedeem) {
+        vault = _vault;
+
+        _disableInitializers();
+    }
+
+    /// @notice Initialize the storage variables stored in the proxy contract.
+    /// The deployer that calls initialize has to approve the ARM's proxy contract to transfer 1e12 liquid assets.
+    /// @param _name The name of the liquidity provider (LP) token.
+    /// @param _symbol The symbol of the liquidity provider (LP) token.
+    /// @param _operator The address of the account that can request and claim Lido withdrawals.
+    /// @param _fee The performance fee that is collected by the feeCollector measured in basis points (1/100th of a percent).
+    /// 10,000 = 100% performance fee
+    /// 1,500 = 15% performance fee
+    /// @param _feeCollector The account that can collect the performance fee
+    /// @param _capManager The address of the CapManager contract
+    function initialize(
+        string calldata _name,
+        string calldata _symbol,
+        address _operator,
+        uint256 _fee,
+        address _feeCollector,
+        address _capManager
+    ) external initializer {
+        _initARM(_operator, _name, _symbol, _fee, _feeCollector, _capManager);
+    }
+
+    /**
+     * @notice Request a withdrawal of oTokens from the Origin Vault.
+     * @param amount The amount of oTokens to withdraw from the Origin Vault.
+     * @return requestId The ID of the Origin Vault withdrawal request.
+     */
+    function requestOriginWithdrawal(uint256 amount) external onlyOperatorOrOwner returns (uint256 requestId) {
+        (requestId,) = IOriginVault(vault).requestWithdrawal(amount);
+
+        // Increase the outstanding withdrawal amount from the Origin Vault
+        vaultWithdrawalAmount += amount;
+
+        emit RequestOriginWithdrawal(amount, requestId);
+    }
+
+    /**
+     * @notice Claim multiple previously requested withdrawals from the Origin Vault.
+     * The caller should check the withdrawal has passed the withdrawal delay
+     * and there is enough liquidity in the Vault.
+     * @param requestIds The request IDs of the withdrawal requests.
+     * @param amountClaimed The total amount claimed across all withdrawal requests.
+     * @return amountClaimed The total amount of oTokens claimed from the Origin Vault.
+     */
+    function claimOriginWithdrawals(uint256[] calldata requestIds) external returns (uint256 amountClaimed) {
+        // Claim the previously requested withdrawals from the Origin Vault.
+        (, amountClaimed) = IOriginVault(vault).claimWithdrawals(requestIds);
+
+        // Store the reduced amount outstanding withdrawals from the Origin Vault
+        vaultWithdrawalAmount -= amountClaimed;
```

Comment:
In case a withdrawal request is transferred to the OriginARM contract, you won't be able to claim it because this line will underflow.

Reply:
This issue also exists in the lido contract.

https://github.com/OriginProtocol/arm-oeth/blob/2e4cffa8548cbd860bdcae9c64a5cbd9a79b2145/src/contracts/LidoARM.sol#L155

A simple fix could be
```
if (vaultWithdrawalAmount <  amountClaimed) {
vaultWithdrawalAmount = 0;
}
```